Week-23/todo-ui/src/pages/AddTodo.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from "react";
+import styles from "./AddTodo.module.less";
+import axios from "axios";
+
+const AddTodo = () => {
+  const [todo, setTodo] = useState("");
+  const [allTodos, setAllTodos] = useState([]);
+  const [todoToUpdate, setTodoToUpdate] = useState(null);
+
+  const getTodo = async () => {
+    try {
+      const data = await axios.get(`http://localhost:3000/api`);
+      console.log("data", data.data);
+      setAllTodos(data.data);
+    } catch (error) {
+      console.error("Error getting todos:", error);
+    }
+  };
+
+  const handleAddTodo = async () => {
+    try {
+      if (todoToUpdate) {
+        // If todoToUpdate is set, update the existing todo
+        await axios.put(`http://localhost:3000/api/${todoToUpdate._id}`, {
+          todo,
+        });
+        setTodoToUpdate(null); // Reset the todoToUpdate state
+      } else {
+        // Otherwise, add a new todo
+        await axios.post("http://localhost:3000/api", {
+          todo,
+        });
+      }
+      console.log("Todo added/updated successfully!");
+      getTodo();
+      setTodo("");
+    } catch (error) {
+      console.error("Error adding/updating todo:", error);
+    }
+  };
+
+  useEffect(() => {
+    getTodo();
+  }, []);
+
+  const handleDelete = async (id) => {
+    console.log("id", id);
+    if (id) {
+      const updatedTodos = allTodos.find((todo) => todo._id !== id);
+      setAllTodos(updatedTodos);
+      await axios
+        .delete(`http://localhost:3000/api/${id}`)
+        .then((res) => {
+          console.log("res", res.data);
+        })
+        .catch((err) => console.log("error", error));
+    }
+  };
+
+  const handleUpdate = async (id) => {
+    console.log("id", id);
+    const todoItem = await allTodos.find((todo) => todo._id == id);
+    if (todoItem) {
+      setTodo(todoItem.todo);
+      setTodoToUpdate(todoItem);
+    }
+  };
```

Comment:
Shouldn’t we also take the update payload as parameter here? Seems like you are using all todos to get current payload, also there’s no update api call being made here, how do we ensure changes persist upon reload?

Reply:
I did not understand why we need update payload here

Reply:
Is it persisting the update upon reload currently?